### DISCUSSION TITLE
Custom styles for lists

### DISCUSSION THREAD

**Discussion by ptram** [jgm/pandoc#9811] [General]
Hi,

I'm trying to convert Markdown to DOCX. By default, it looks like all lists (ordered or unordered) have the "Compact" paragraph style applied.

I would like to apply my own custom styles. In the source code I have things like:

```
::: {custom-style="Numbered 1"}
1. This is the first item in a numbered lists.
:::

::: {custom-style="Numbered"}
2. This is the second item.
3. And this is the third item.
:::

```

Unfortunately, all the lists have still the "Compact" paragraph style assigned, with list properties for continued or restarting numbering.

Is there a way to make my custom styles override the default one?

**Comment by iandol** [comment]
Does it need to be paragraph styles, i.e. if the items need to be styled could inline styles work?

```pandoc
1. [This is the first item in a numbered lists.]{custom-style="AlertTok"}
2. [This is the second item.]{custom-style="AnnotationTok"}
3. [And this is the third item.]{custom-style="AnnotationTok"}
```

**Reply by ptram** [reply]
Thank you Ian. Unfortunately, it has to be a paragraph style, because it is not just a decoration, but a structural matter.

Having the first item as a "Numbered 1" paragraph would allow considering it as the paragraph with the first number in the list. This property is among the paragraph properties in page layout programs.

Also, this would allow the design of the spacing and tab positioning for that paragraph.

Since the numbering and bullet style is among the paragraph style properties, it would also be necessary to have separate styles for Numbered and Bulleted lists, with the different numbering styles.

Be aware that I'm thinking to the DOCX styles as an intermediate format for page layout programs. This is the final target of most word processor documents, but their logic is not always overlapping.

**Reply by iandol** [reply]
I think @massifrg's syntax is working, the DOCX has those paragraph styles attached. I suspect it should be possible to use a lua filter (or certainly Scrivener style/replacement) to automate this...

![Screenshot 2024-05-30 at 09 21 00](https://github.com/jgm/pandoc/assets/153905/3cd83d34-752b-4e9c-a4e0-9058b1084df1)

**Comment by massifrg** [comment]
```markdown
1.  ::: {custom-style="Numbered 1"}
    This is the first item in a numbered lists.
    :::

2.  ::: {custom-style="Numbered"}
    This is the second item.
    :::

3.  ::: {custom-style="Numbered"}
    And this is the third item.
    :::
```

I tried to convert this into HTML, native and DOCX, and it seems to work.

Unfortunately, you need to repeat the Div with custom-style on every list item.

**Comment by iandol** [comment]
@ptram — I wrote a quick lua filter that takes bullet or ordered lists and wraps them in divs (it builds the AST @massifrg described). 

### So with this command & markdown:

`pandoc -L enstyle-list.lua -o out.docx`

```markdown
* foo foo
* bar bar

Text.

1. foo foo
2. bar bar
```

### You get this in Word:

![Screenshot 2024-05-30 at 12 56 18](https://github.com/jgm/pandoc/assets/153905/f2a234dd-6fb7-45ab-a629-9aa453010561)

### The filter: 

https://github.com/iandol/dotpandoc/blob/master/filters/enstyle-list.lua

**Reply by ptram** [reply]
@iandol and @massifrg, thank you very much! The filter works great, and even removes the hassle of having to apply the custom paragraph style. It really changes everything!

**Reply by iandol** [reply]
👍 Pandoc filters are wonderful indeed; if you notice the code itself is really simple, and almost any element of a document can be tweaked like this. You can add filters for each particular detail to really dial in you final document…
